internal/image/url.go

```diff
@@ -0,0 +1,150 @@
+package image
+
+import (
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"net/url"
+	"path"
+	"regexp"
+	"time"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/golang-jwt/jwt/v4"
+	"github.com/kubev2v/migration-planner/internal/store/model"
+	"github.com/pkg/errors"
+)
+
+var jwtPayloadRegexp = regexp.MustCompile(`^.+\.(.+)\..+`)
+
+type payload struct {
+	Sub string `json:"sub"` // used by OCM tokens
+}
+
+const (
+	// ImageExpirationTime define the expiration of the image download URL
+	ImageExpirationTime = 4 * time.Hour
+)
+
+func GenerateShortImageDownloadURLByToken(baseUrl string, source *model.Source) (string, *strfmt.DateTime, error) {
```

Comment:
I'm not nit-picking here but the func name could be much simpler. What does "ShortImage" mean in this context? :)

Reply:
Well `ShortImage` look bit too abstract. What about `DownloadImageUrl`